packages/core/src/InstanceSettings.ts

```diff
@@ -61,6 +71,22 @@ export class InstanceSettings {
 	 */
 	instanceRole: InstanceRole = 'unset';
 
+	/**
+	 * Transient ID of this n8n instance, for scaling mode.
+	 * Reset on restart. Do not confuse with `instanceId`.
+	 *
+	 * @example 'main-bnxa1riryKUNHtln'
+	 * @example 'worker-nDJR0FnSd2Vf6DB5'
+	 * @example 'webhook-jxQ7AO8IzxEtfW1F'
+	 */
+	hostId = '';
+
+	setHostId() {
+		strict(this.hostId === '', 'Expected `hostId` to be unset');
+
+		this.hostId = `${this.instanceType}-${nanoid()}`;
+	}
```

Comment:
Maybe add a test for this?

Reply:
[e6f0382](https://github.com/n8n-io/n8n/pull/11262/commits/e6f038238b0e8ec4214b82b8f8f764ef5378456c)